cloudify_aws/ec2/elasticip.py

```diff
@@ -0,0 +1,300 @@
+########
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+# Third-party Imports
+from boto import exception
+
+# Cloudify imports
+from cloudify_aws import utils, constants
+from cloudify import ctx
+from cloudify.exceptions import NonRecoverableError
+from cloudify.decorators import operation
+from cloudify_aws.base import AwsBaseNode, AwsBaseRelationship
+
+
+@operation
+def creation_validation(**_):
+    return ElasticIP().creation_validation()
+
+
+@operation
+def create(args=None, **_):
+    return ElasticIP().created(args)
+
+
+@operation
+def delete(args=None, **_):
+    return ElasticIP().deleted(args)
+
+
+@operation
+def associate(args=None, **_):
+    return ElasticIPInstanceConnection().associated(args)
+
+
+@operation
+def disassociate(args=None, **_):
+    return ElasticIPInstanceConnection().disassociated(args)
+
+
+class ElasticIPInstanceConnection(AwsBaseRelationship):
+
+    def __init__(self, client=None):
+        super(ElasticIPInstanceConnection, self).__init__(client=client)
+        self.not_found_error = 'InvalidAssociationID.NotFound'
+        self.resource_id = None
+        self.source_get_all_handler = {
+            'function': self.client.get_all_instances,
+            'argument':
+                '{0}_ids'.format(constants.INSTANCE['AWS_RESOURCE_TYPE'])
+        }
+
+    def associate(self, args=None, **_):
+        """ Associates an Elastic IP created by Cloudify with an EC2 Instance
+        that was also created by Cloudify.
+        """
+
+        instance_id = \
```

Comment:
all this should be taken care of in 
https://github.com/cloudify-cosmo/cloudify-aws-plugin/blob/master/cloudify_aws/base.py#L152